### ISSUE TITLE
Update alert steals focus while typing in master password

### ISSUE THREAD

**Issue by kyldvs** [bitwarden/clients#2608]
## Describe the Bug

After launching the desktop app I will immediately begin typing in my master password. If there is an update, while I'm typing, the alert pops up and takes the focus away from the text field I was currently typing in. This often causes me to perform unintended actions without having a chance to even read the alert.

## Steps To Reproduce

1. Launch app that needs an update
2. Start typing in master password field
3. See update alert steal focus

## Expected Result

The update alert should not steal focus, not trigger while I'm typing, or wait until I've entered my master password.

## Actual Result

The update alert steal focus while I'm typing and causes unintended actions to be taken.

## Environment

- Operating system: Windows 10
- Installation method: Downloaded from Bitwarden.com
- Build Version: A version prior to 1.26.3, I don't know the exact version due to the bug itself, but it has been happening consistently for a long time

## Additional Context

I really like Bitwarden but this one issue is becoming increasingly frustrating every single time I launch the app. I would love some change to this alert to fix the issue

**Comment by oopoopoop** [comment]
What if we just block the update dialog's buttons for say 5 seconds before accepting inputs again?
Like one of those buttons with a countdown on them? i.e. "(5s)"

This should:
a) prevent the Enter key from accepting the update
b) give the user enough time to properly respond to the dialog

I certainly did not know that it would even steal focus from other applications (even for full screen ones), because I always start Bitwarden only when I need it and it will have my full attention anyways. I don't mind it prompts me with the latest update ASAP and tries to stop me from finishing my password. If I find the update worthwhile, this will end up saving me some time typing my full password twice!

If this simply fix is acceptable, I wish this to be fixed ASAP.
We might open another issue for the "stealing (other application's) focus" part of the issue.

**Comment by parski** [comment]
I would prefer it not prompt at all when I'm typing in my password.

**Comment by bhughes339** [comment]
I'd like to suggest (as I did in the duplicate ticket) utilizing the Windows notification system. Here's an example of a well-designed update notification (PowerToys by Microsoft):
![image](https://user-images.githubusercontent.com/16418739/190826907-e65d7f9a-792a-4472-8455-4ea48b7c6f14.png)

**Comment by teunvg** [comment]
Any updates on this?

It is the single most annoying thing about Bitwarden that almost every time I type my (very long) master password the update dialog pops up halfway through, one of my keystrokes triggers the update option, and it closes the application.

**Comment by 0yz** [comment]
How is this still not fixed? This is completely embarrassing and should've never been implemented like this in the first place.

**Comment by chocolatechipcats** [comment]
The update notification also steals focus when minimized to tray. Has caused issues when I'm playing fullscreen games, some of which do _not_ alt+tab gracefully. Would prefer an OS-native notification.

**Comment by 0verEngineer** [comment]
This problem annoys me since i use Bitwarden, the update should not be a pupup notification, it should only be a yellow bar on top of the app or sth. like that.

**Comment by wrffrz** [comment]
This bug hits me literally every time there's an update.  I don't keep bitwarden running in the background and the only time I start the application is when I want to look up a password right then, so I'll start typing my master password only to have the application disappear and the UAC prompt show up because the update dialog popped up while I was typing causing me to curse the developers for the minute or two while it updates before I can get the password I was trying to look up.

**Comment by kyldvs** [comment]
@eliykat would it be possible to get an update on the status of this since it's been 2 years and this is still a common frustration among users?

**Comment by gtflip** [comment]
This just happened to me again with the Windows client.  Please fix this.

I was annoyed enough to use Google to find this issue, since github issue search seemed to ignore the quotes.

**Comment by jtomkiew-mng** [comment]
Boop, any updates?

**Comment by dirk-tf** [comment]
This issue is not treated with the urgency it need to. I also have spaces and had my master password end up "elsewhere".

PSA for now:
NEVER log into you bitwarden while attached to any kind of screen sharing (MS Teams, projector for presentation, etc).

**Comment by teynar** [comment]
Same issue here, please fix it, it is quite frustrating to start typing your password and having to rewrite it all over again cause the focus got lost, but the fact that it may end up somewhere else is inherently dangerous.

Maybe a text underneath saying Checking for updates... with a spinner that would change to either: No updates available / Update to x.y.z would be better, for example.

**Comment by pamperer562580892423** [comment]
@trmartin4 Does it have any relevance, that https://github.com/bitwarden/clients/pull/16526 is not merged as well (yet)?

**Comment by dani-garcia** [comment]
@pamperer562580892423 No, that was just a temporary PR to get some builds for QA testing. 

The fix for this (#15606) should be available in the next release in a couple of weeks, and it would start taking effect once it's time to update to the release after that.
